sql-plugin/src/main/spark350/scala/com/nvidia/spark/rapids/shims/SparkShims.scala

```diff
@@ -22,15 +22,33 @@ package com.nvidia.spark.rapids.shims
 
 import com.nvidia.spark.rapids._
 
-import org.apache.spark.sql.catalyst.expressions.{Expression, PythonUDAF, ToPrettyString}
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.{AttributeReference, Expression, PythonUDAF, ToPrettyString}
 import org.apache.spark.sql.execution.SparkPlan
 import org.apache.spark.sql.execution.adaptive.TableCacheQueryStageExec
 import org.apache.spark.sql.execution.columnar.InMemoryTableScanExec
+import org.apache.spark.sql.execution.datasources.{FilePartition, FileScanRDD, HadoopFsRelation, PartitionedFile}
 import org.apache.spark.sql.execution.window.WindowGroupLimitExec
 import org.apache.spark.sql.rapids.execution.python.GpuPythonUDAF
-import org.apache.spark.sql.types.StringType
+import org.apache.spark.sql.types.{StringType, StructType}
 
 object SparkShimImpl extends Spark340PlusNonDBShims {
+  override def getFileScanRDD(
+      sparkSession: SparkSession,
+      readFunction: PartitionedFile => Iterator[InternalRow],
+      filePartitions: Seq[FilePartition],
+      readDataSchema: StructType,
+      relation: Option[HadoopFsRelation],
+      metadataColumns: Seq[AttributeReference] = Seq.empty): RDD[InternalRow] = {
+    if (relation.isDefined) {
+      new FileScanRDD(sparkSession, readFunction, filePartitions, readDataSchema, metadataColumns,
+        metadataExtractors = relation.get.fileFormat.fileConstantMetadataExtractors)
+    } else {
+      new FileScanRDD(sparkSession, readFunction, filePartitions, readDataSchema, metadataColumns)
+    }
```

Comment:
Typically in Scala calling `.get` on an `Option` is an anti-pattern.

```suggestion
    new FileScanRDD(sparkSession, readFunction, filePartitions, readDataSchema, metadataColumns,
        relation.map(_.fileFormat.fileConstantMetadataExtractors).getOrElse(Map.empty))
```